packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -740,19 +740,15 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
 
     if (_needToReportFirstFrame && _reportFirstFrame) {
       assert(!_firstFrameCompleter.isCompleted);
-      // TODO(liyuqian): use a broadcast stream approach
-      final TimingsCallback oldCallback = WidgetsBinding.instance.window.onReportTimings;
-      WidgetsBinding.instance.window.onReportTimings = (List<FrameTiming> timings) {
+      StreamSubscription<FrameTiming> firstFrameSubscription;
+      firstFrameSubscription = WidgetsBinding.instance.frameTimingStream.listen((FrameTiming timing) {
```

Comment:
Would Stream.first be appropriate here since you cancel the stream after you get the first even right away?

https://master-api.flutter.dev/flutter/dart-async/Stream/first.html

Reply:
Added more details to the PR description.

Also added test which verifies that the broadcast stream won't be shut down (cancelled) by `Stream.first`. (`Stream.first` will only cancel its own subscription, and leave other subscription still running.)

Reply:
So, we should be able to use it here as well?

Reply:
Yes. Thanks!